### PR TITLE
Spanish translation of ch06-03-if-let.md

### PR DESCRIPTION
**Pull request by dbucci24**
full translation of the file ch06-03-if-let.md

-599 words

### CODE REVIEW

File: second-edition/src/ch06-03-if-let.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
-## Concise Control Flow with `if let`
+## Flujo de control conciso con `if let`
+
+La sintaxis `if let` te permite combinar `if` y `let` de una manera menos verbosa
```
Comment (thecodix) [inline second-edition/src/ch06-03-if-let.md]: verbose: detallado/a

File: second-edition/src/ch06-03-if-let.md [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
+para manejar valores que coincide con un patrón e ignora el resto. Considera el programa
```
Comment (thecodix) [inline second-edition/src/ch06-03-if-let.md]: coinciden

File: second-edition/src/ch06-03-if-let.md [hunk 3]
```diff
@@ -13,16 +14,17 @@ match some_u8_value {
...
-`Some<u8>` value or the `None` value. To satisfy the `match` expression, we
-have to add `_ => ()` after processing just one variant, which is a lot of
-boilerplate code to add.
+<span class="caption">Listado 6.6: Un `match` que solo se preocupa de ejecutar
+código cuando el valor es `Some(3)`</span>
+
+Queremos hacer algo con la coincidencia `Some(3)` pero no hacemos nada con ningún
+otro valor `Some<u8>` o el valor `None`. Para satisfacer la expresión `match`, tenemos
+que agregar `_ => ()` después de procesar solo una variante, que es mucho código
+repetitivo para agregar.
```
Comment (thecodix) [inline second-edition/src/ch06-03-if-let.md]: please leave boilerplate until we find a better word for it

File: second-edition/src/ch06-03-if-let.md [hunk 4]
```diff
@@ -31,26 +33,26 @@ if let Some(3) = some_u8_value {
...
 
-Using `if let` means you have less to type, less indentation, and less
-boilerplate code. However, we’ve lost the exhaustive checking that `match`
-enforces. Choosing between `match` and `if let` depends on what you’re doing in
-your particular situation and if gaining conciseness is an appropriate
-trade-off for losing exhaustive checking.
+El uso de `if let` significa que tienes menos para escribir, menos indexáción, y
+menos código repetido. Sin embargo, hemos perdido la comprobación exhaustiva que `match`
+ejecuta. Eligiendo entre `match` e `if let` depende de que estes haciendo en
+tu situación particular y si gana concisidad es un intercambio por perder una
```
Comment (thecodix) [inline second-edition/src/ch06-03-if-let.md]: concisidad : precision

File: second-edition/src/ch06-03-if-let.md [hunk 5]
```diff
@@ -31,26 +33,26 @@ if let Some(3) = some_u8_value {
...
-`Coin` enum definition in Listing 6-4, where the `Quarter` variant also held a
-`UsState` value. If we wanted to count all non-quarter coins we see while also
-announcing the state of the quarters, we could do that with a `match`
-expression like this:
+Podemos incluir un `else` con un `if let`. El código de bloque que va con el
+`else` es el mismo bloque de código que podría ir con el caso `_` en la expresión
+`match` que es equivalente al `if let` y `else`. Recuerda la definición de enumeración
+`Coin` en el Listado 6-4, donde la variante `Quarter` también tenía un valor
+`UsState`. Si quisieramos contar todas las monedas que no son cuartos vemos también
+mientras anuncioamos el estado de los cuartos, podríamos hacer eso con una expresión `match`
```
Comment (thecodix) [inline second-edition/src/ch06-03-if-let.md]: anunciamos

**Comment by thecodix** [comment]
Hey, apart from the comments, there is some amount of typos, please remember to run a spell checker before putting the code for merge

Thanks again!

**Comment by dbucci24** [comment]
Yes, they are typing errors by writing fast haha, sorry, they are fixed!

**Comment by thecodix** [comment]
No problem, thanks

**Pull request merged**